packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -531,6 +530,31 @@ class FlutterValidator extends DoctorValidator {
     );
   }
 
+  ValidationMessage _getFlutterVersionMessage(String frameworkVersion, String versionChannel) {
+    final String flutterVersionMessage = _userMessages.flutterVersion(frameworkVersion, versionChannel, _flutterRoot());
+
+    if (versionChannel == 'unknown' || frameworkVersion == '0.0.0-unknown') {
```

Comment:
Out of curiosity, when is this value `0.0.0-unknown`? And can you add a comment with the answer so I don't ask you next time :)